Dockerfile

```diff
@@ -1,5 +1,5 @@
 # For installing npm dependencies and building static css/js
-FROM node:10-alpine
+FROM node:16-alpine3.14
```

Comment:
True that 16 is not on LTS yet, it will be. So 16 is most recent release, intended for LTS, with the longest support lifespan. And this is only used a build tool, so honestly the version matters less, what matters most is that it is supported and that we won't have to change it for a while.

14 is the current LTS, but will soon be in maintenance. It could be good to be on a maintenance release, for reduced risk. For us, this is just for development and build, so I don't think the risks are high. Choosing a later release will just move the goalposts further.

Good suggestion on the [engines].

Reply:
Added engines selecting node 16, and also `engine-strict`, which makes the package.json compatible with npm.
`yarn` 1 only has `engine-strict` enabled if engines are used. `yarn` 2 relaxes this; I don't know if it uses `engine-strict`, but probably does as this caused problems for some users.

This change is nice for consistency, as it ensures that only node 16 will always be used with the project.

We don't use node-16-only features ourselves afaik, but I've found that some libraries can come to depend on features of more recent node releases, and are often tested only with more recent releases. Certainly our previous usage of node 10 would be problematic, as it was out of support and I doubt many libraries test against it.

Reply:
Bah, now github actions refuses to run node 16.
https://github.com/kbaseIncubator/dashboard-redesign/pull/88/checks?check_run_id=3478619705#step:3:6

I think that has to do with this line in the github action definition:
https://github.com/kbaseIncubator/dashboard-redesign/blob/main/.github/workflows/test.yml#L11

The following link indicates that, today, `ubuntu-latest` refers to 20.04:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

So we need to either use >=14.17.5 for now or update node in the github action. If it were me, I would move to 14.17.5 and update the github `runs-on` to be `ubuntu-20.04`. I'm not sure if it is better to pin the engine to 14.17.5 (since that's what the CI will always be running on) or make it `>=` to be more flexible.

Reply:
Nice, I was not sure how easy it would be to specify the node version. Fortunately it was easy! Thanks 👍